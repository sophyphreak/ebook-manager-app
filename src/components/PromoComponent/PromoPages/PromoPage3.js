import React, { Component } from 'react';
import {
  Card,
  CardHeader,
  CardBody,
  Button,
  Form
} from 'reactstrap';

import Description from '../../FormElements/Description';
import AuthorBio from '../../FormElements/AuthorBio';

const PromoPage3 = ({
  error,
  onBack,
  onSubmit,

  description,
  onDescriptionChange,

  authorBio,
  onAuthorBioChange
}) => (
    <Card className="animated fadeIn">
      <CardHeader>
        <h4><strong>Book Promo - 3 of 3</strong></h4>
      </CardHeader>
      <CardBody>
        <Form onSubmit={onSubmit}>
          <Description
            value={description}
            onChange={onDescriptionChange}
          />
          <AuthorBio
            value={authorBio}
            onChange={onAuthorBioChange}
          />
          {error && <p>{error}</p>}
          <Button
            onClick={onBack}
            size="sm"
            color="basic"
          >
            Go Back
          </Button>
          <Button
            type="submit"
            size="sm"
            color="primary"
            value="Submit"
          >
            Next Section
          </Button>
        </Form>
      </CardBody>
    </Card>
  );

export default PromoPage3;