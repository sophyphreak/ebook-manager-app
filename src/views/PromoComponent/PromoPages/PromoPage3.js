import React, { Component } from 'react';
import { Card, CardHeader, CardBody, Button, Form } from 'reactstrap';

import Description from '../../FormElements/Description';
import AuthorBio from '../../FormElements/AuthorBio';

const PromoPage3 = ({
  error,
  onBack,
  onSubmit,

  description,
  onDescriptionChange,

  authorBio,
  onAuthorBioChange
}) => (
  <Card className="animated fadeIn">
    <CardHeader>
      <h4>
        <strong>Book Promo - 3 of 3</strong>
      </h4>
    </CardHeader>
    <CardBody>
      <Form onSubmit={onSubmit}>
        <Description
          label="Promotional Book Description (Max 2,000 characters)*"
          description={description}
          onDescriptionChange={onDescriptionChange}
          hasError={!!error.message}
          errorMessage={error.description}
        />
        <AuthorBio
          label="Author Biography:*"
          authorBio={authorBio}
          onAuthorBioChange={onAuthorBioChange}
          hasError={!!error.message}
          errorMessage={error.authorBio}
        />
        {error.message && <p>{error.message}</p>}
        <Button
          className=".back-button"
          onClick={onBack}
          size="sm"
          color="basic"
        >
          Go Back
        </Button>
        <Button type="submit" size="sm" color="primary" value="Submit">
          Submit
        </Button>
      </Form>
    </CardBody>
  </Card>
);

export default PromoPage3;
