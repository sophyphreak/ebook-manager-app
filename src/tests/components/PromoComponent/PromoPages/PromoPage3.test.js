import React from 'react';
import { shallow } from 'enzyme';
import PromoPage3 from '../../../../components/PromoComponent/PromoPages/PromoPage3';

let wrapper;

beforeEach(() => {
  wrapper = shallow(<PromoPage3 />);
});

test('should render PromoPage3 correctly', () => {
  expect(wrapper).toMatchSnapshot();
});