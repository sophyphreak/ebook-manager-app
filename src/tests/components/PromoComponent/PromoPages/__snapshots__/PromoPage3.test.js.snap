// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render PromoPage3 correctly 1`] = `
<Card
  className="animated fadeIn"
  tag="div"
>
  <CardHeader
    tag="div"
  >
    <h4>
      <strong>
        Book Promo - 3 of 3
      </strong>
    </h4>
  </CardHeader>
  <CardBody
    tag="div"
  >
    <Form
      tag="form"
    >
      <Description
        errorMessage=""
        hasError={false}
        label="Promotional Book Description (Max 2,000 characters)*"
      />
      <AuthorBio
        errorMessage=""
        hasError={false}
        label="Author Biography:*"
      />
      <Button
        className=".back-button"
        color="basic"
        size="sm"
        tag="button"
      >
        Go Back
      </Button>
      <Button
        color="primary"
        size="sm"
        tag="button"
        type="submit"
        value="Submit"
      >
        Submit
      </Button>
    </Form>
  </CardBody>
</Card>
`;
